doc/stacker_yaml.md

```diff
@@ -11,20 +11,28 @@ processed in stacker as:
 
     1 2 3
 
-That is, variables of the form `$FOO` or `${FOO}` are supported, and variables
-with `${FOO:default}` a default value will evaluate to their default if not
-specified on the command line. It is an error to specify a `${FOO}` style
-without a default; to make the default an empty string, use `${FOO:}`.
+In order to avoid conflict with bash or POSIX shells in the `run` section, 
+only placeholders with two braces are supported, e.g. `${{FOO}}`.
+Placeholders with a default value like `${{FOO:default}}` will evaluate to their default if not
+specified on the command line or in a substitution file.
 
-In addition to substitutions provided on the command line, the following
-variables are also available with their values from either command
+It is an error to specify a `${{FOO}}` placeholder without a default; to make
```

Comment:
Wouldn't it be just as robust and more intuitive to have ${{FOO}} work fine when FOO is set, and to simply fail (with a good error message) if FOO substitution is not defined?  Then ${{FOO:}} would still be needed if you want to default to empty, but i don't believe that's often the intent.

Reply:
That's the behavior you get now. If you have `${{FOO}}` and you don't provide `--substitution FOO=whatever`, you will get the error `No value for substitution FOO`.

Reply:
(so this does need to be worded more clearly, I'll do that)

Reply:
(done)